tests/kafkatest/services/streams.py

```diff
@@ -353,6 +353,10 @@ class StreamsSmokeTestJobRunnerService(StreamsSmokeTestBaseService):
     def __init__(self, test_context, kafka):
         super(StreamsSmokeTestJobRunnerService, self).__init__(test_context, kafka, "process")
 
+class StreamsSmokeTestEOSJobRunnerService(StreamsSmokeTestBaseService):
+    def __init__(self, test_context, kafka):
+        super(StreamsSmokeTestEOSJobRunnerService, self).__init__(test_context, kafka, "process-eos")
```

Comment:
Let me know if there's a better way to do this...

Reply:
I thought there were already existing`EOS` tests that use the `StreamsSmokeTest`?

Reply:
To my other comment above: what's the difference of running `StreamsSmokeTestBaseService#process-eos` v.s. `StreamsEosTestBaseService#process`? The former uses StreamsSmokeTest while the latter use StreamsEosTest client. Can we just consolidate them?

Reply:
Shameful admission: I thought about this, but just didn't take the time to check.

The bounce test was obviously duplicated, but the eos test is more complex. I knew I'd have to do a case-by-case check to make sure I didn't remove coverage. I'll do my homework now...

Reply:
Ok, so the smoke test and the eos test are similar, but not identical.

The smoke test application has more features in it, though. So, we have more feature coverage under eos when we test with the smoke test.

The eos test evaluates two topologies one with no repartition, and one with a repartition. The smoke test topology contains repartitions, so it only tests _with_ repartition. I think that it should be sufficient to test only _with_ repartition.

The eos test verification specifically checks that "all transactions finished" (`org.apache.kafka.streams.tests.EosTestDriver#verifyAllTransactionFinished`). I'm not clear on exactly what we're looking for here. It looks like we create a transactional producer and send a record to each partition and then expect to get all those records back, without seeing any other records. But I'm not sure why we're doing this. If we want to drop the eos test, then we might need to add this to the smoke test verification.

WDYT?

Reply:
`verifyAllTransactionFinished` aimed to avoid a situation that some dangling txn is open forever, without committing or aborting. Because the consumer needs to guarantee offset ordering when returning data, with read-committed they will also be blocked on those open txn's data forever (this usually indicates a broker-side issue, not streams though, but still).

I think we should still retain this check if we want to merge in the EosTest to SmokeTest, but also if you do not want to drag on it we could keep it as a separate ticket and merge this as-is. Your call :)

Reply:
Ok, thanks for the explanation. Let's merge this, and I'll create a ticket to follow up by including this check and removing the eos test.

That'll help review burden as well, I think.